node/controllers/channelPurpose/index.js

```diff
@@ -0,0 +1,212 @@
+              where: { email: mattermostUser.email },
+              include: [
+                {
+                  model: Interest,
+                  as: 'interests',
+                  attributes: ['id', 'name'],
+                  through: { attributes: [] },
+                },
+              ],
+            })
+              .then(fetchedUsers => {
+                return fetchedUsers.interests
+              })
+              .catch(() => {
+                return undefined
+              })
+          }
+        )
+        // Wait for the promises to resolve
+        const usersInterests = await Promise.all(usersInterestsPromises)
+        // Create empty object we are going to populate with the interests and set to be the purpose of the channel
+        const allInterests = {}
+
+        // Parse through the resolved promises
+        usersInterests.forEach(interests => {
+          // Each promise contains a list of interests, add them to the allInterests object
+          // If user was not found (is undefined), do nothing
+          if (interests) {
+            interests.forEach(interest => {
+              allInterests[interest.name] =
+                interest.name in allInterests
+                  ? (allInterests[interest.name] += 1)
+                  : 1
+            })
+          }
+        })
+        // Update mattermost channel purpose to match its users interests
+        await axios.put(`${mattermostUrl}/channels/${channel.id}/patch`, {
+          purpose: JSON.stringify(allInterests),
+        })
+      }
+    })
+    return res.status(200).send({
+      success: true,
+      message: 'did it',
+    })
+  } catch (error) {
+    return res.status(500).send({
+      success: false,
+      message: 'Error',
+      error,
+    })
+  }
+}
+
+// This function adds given users interests to the channel purpose
+export const addUserInterestsToPurpose = async (req, res) => {
+  // Set bearer token
+  axios.defaults.headers.common.Authorization = `Bearer ${
+    process.env.MASTER_TOKEN
+  }`
+
+  // Get channel id from request parameters
+  const { channelId } = req.params
+  // Get user from passport set user object
+  const userId = req.user.dataValues.id
+
+  try {
+    // Make promises that return given mm channel and user interests.
+    const channelPromise = axios.get(`${mattermostUrl}/channels/${channelId}`)
+    const userPromise = User.findOne({
+      where: { id: userId },
+      include: [
+        {
+          model: Interest,
+          as: 'interests',
+          attributes: ['id', 'name'],
+          through: { attributes: [] },
+        },
+      ],
+    }).then(fetchedUser => {
+      return fetchedUser.interests
+    })
+    // Wait for both of the promises to complete
+    const completedPromises = await Promise.all([channelPromise, userPromise])
+    // Get current channel purpose and make it into js object
+    const channelPurpose = JSON.parse(completedPromises[0].data.purpose)
```

Comment:
Maybe same JSON.parse problem here. I did not run into any problems, but better make sure no errors are induced.